pkg/piecestore/pstore.go

```diff
@@ -0,0 +1,152 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package pstore
+
+import (
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/aleitner/FilePiece"
+	"github.com/zeebo/errs"
+)
+
+// Errors
+var (
+	ArgError = errs.Class("argError")
+	FSError  = errs.Class("fsError")
+)
+
+// PathByHash -- creates datapath from hash and dir
+func PathByHash(hash, dir string) (string, error) {
+	if len(hash) < 20 {
```

Comment:
you should move most of your hash and dir checking to this method so you don't need to duplicate it in all of the other functions.

overall, without playing "code golf", the less lines of code, the better